### PR TITLE
Filter pills for Client Portfolio filters and ability to clear/reset them all

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
### Edge-Cases Galore
**Proxy CDO View:**
- [x] If x on pill: deselect CDO in dropdown.
- [x] If deselect all: reset to current CDO (unfortunately, deselect all in the dropdown keeps currentCDO as well, they would have to click on the actual pill or in picky to deselect).

**Bid Season:**
- [x] If x on pill: deselect Season in dropdown.
- [x] 	if last selected Season: remove Filter By pill (don’t reset Filter By state, just remove the pill and filtering for it)
- [x] If deselect all: reset to none selected
- [x] add select all to picky (does not toggle to deselect all tho)

**Filter By:**
- [x] If x on pill: reset to All
- [x] If deselect all: reset to All and make sure disabled
- [x] Only 1 pill at a time, reset pill value on update of dropdown value.
- [x] Remove/Do not Show pill if default value of All
- [x] IF no Bid Season pill, then don’t show/remove; on the other hand, when Bid Season is selected, default add a pill for Filter By(only if one was there before).

Final check:
- [x] query properly updating for all dropdowns and scenarios

-----Won't be showing pill for Display Clients 😭 -----------

**Display Clients:**
- [x] If x on pill: reset to 5
- [x] If deselect all: reset to 5
- [x] Only 1 pill at a time, reset pill value on update of dropdown value.
- [x] Remove/Do not Show pill if default value of 5

-----Won't be showing pill for Sort by 😭 -----------
**Sort By:**
- [x] If x on pill: reset to Last Name: A-Z
- [x] If deselect all: reset to Last Name: A-Z
- [x] Only 1 pill at a time, reset pill value on update of dropdown value.
- [x] Remove/Do not Show pill if default value of Last Name: A-Z

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
Pills properly generating when just using the dropdowns. Properly removing pill when default selected in dropdowns.

To Do: remove pills when clicked.
Remove all pills and reset to defaults when Clear All selected

**Comment by elizabeth-jimenez** [comment]
two last annoying issues:
~ resetting of Filter By does not update to 'All'
~ when Leah goes to page, she is selected in proxy CDO dropdown by default, so i prop-drilled cdo (changed from cdoLength) hoping to be able to grab and 'set' Leah from there. The problem isn't so much generating the pill, it is when we hit Deselect All but proxyCDO does not update, bc it is set up to only do so when the local state changes(which it doesn't bc it was never set with the default of Leah).

File: src/Components/BidderPortfolio/BidControls/BidControls.jsx [hunk 1]
```diff
@@ -70,7 +70,9 @@ class BidControls extends Component {
   generatePills = () => {
     const pills = [];
     this.state.proxyCdos.forEach(a => {
-      pills.push({ description: a.name, selectionRef: 'proxyCdos', codeRef: a.hru_id });
+      if (!isEqual(a, '')) {
+        pills.push({ description: a.name, selectionRef: 'proxyCdos', codeRef: a.hru_id });
+      }
```
Comment (mjoyce91) [inline src/Components/BidderPortfolio/BidControls/BidControls.jsx]: What does this do?

Comment (elizabeth-jimenez) [reply src/Components/BidderPortfolio/BidControls/BidControls.jsx]: that's the hack :( i wanted to ask you and scott about it lol

File: src/Components/BidderPortfolio/BidControls/BidCyclePicker/BidCyclePicker.jsx [hunk 2]
```diff
@@ -74,7 +77,15 @@ class BidCyclePicker extends Component {
...
-    return this.bidSeasonsToIds();
+    const { arrayValue } = this.state;
+    const { seasons } = this.props;
+    const ids$ = isArray(seasons) ? [...seasons] : [];
+    return filter(ids$, f => indexOf(arrayValue, f.description) > -1);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  setMultipleOptionFromParent(seasonObjs) {
+    this.setState({ arrayValue: flatMap(seasonObjs, a => a.description) }, () => this.setSeasons());
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/BidControls/BidCyclePicker/BidCyclePicker.jsx:88]: ❤️  a lodash beaut

Comment (mjoyce91) [reply src/Components/BidderPortfolio/BidControls/BidCyclePicker/BidCyclePicker.jsx:88]: I think regular `map` would work here as well?

Comment (elizabeth-jimenez) [reply src/Components/BidderPortfolio/BidControls/BidCyclePicker/BidCyclePicker.jsx:88]: You don't want lodash?! Maybe, not sure? This is taking just the string and returning an array of strings to arrayValue. BidCyclePicker is weird

Comment (elizabeth-jimenez) [reply src/Components/BidderPortfolio/BidControls/BidCyclePicker/BidCyclePicker.jsx:88]: 🤔  but im not sure, def open to learning and changing if you think regular map is better

Comment (elizabeth-jimenez) [reply src/Components/BidderPortfolio/BidControls/BidCyclePicker/BidCyclePicker.jsx:88]: discussed in Slack: change to `seasonObjs.map(a => a.description) ` woot. Thanks Mike!

**Comment by elizabeth-jimenez** [comment]
Couldnt figure out these last 2 edge cases:
Filter By: 
 If x on pill: reset to All
 If deselect all: reset to All and make sure disabled

**Comment by elizabeth-jimenez** [comment]
@mjoyce91 open to the mapping solution you said could be implemented for the empty string `value: ''` issue

**Comment by mjoyce91** [comment]
Screen Shot 2020-10-14 at 10 28 23 AM

I think I'd:
- move the Clear Filters link above the filters
- add some padding between the X and the text

**Comment by mjoyce91** [comment]
If you click on a client profile and then come back, the filters are persisted but the pills don't appear

**Comment by mjoyce91** [comment]
>open to the mapping solution you said could be implemented for the empty string value: '' issue. Curious where that would happen 🤔

@elizabeth-jimenez Let's come back to that, I forget exactly how everything works together

**Comment by elizabeth-jimenez** [comment]
> Screen Shot 2020-10-14 at 10 28 23 AM
> 
> I think I'd:
> 
> * move the Clear Filters link above the filters
> * add some padding between the X and the text

haha i knew you wouldnt like that! Neither do I! I spent a few hours trying to do so without changing the actual components, but it was a no. Bc i'm using those components and they have classes already in them like `usa-grid-full` (ex ResultsPillContainer), i'd have to modify those components to receive unique classes. Which i can, it just felt a bit hacky

**Comment by elizabeth-jimenez** [comment]
> If you click on a client profile and then come back, the filters are persisted but the pills don't appear

Ah, i think that was partially intentional, bc we don't want to generate pills the first time they land on client profile, like default Leah pill. I guess this feature was more applicable when the Display Clients and Sort By pill functionality remained

**Comment by elizabeth-jimenez** [comment]
> If you click on a client profile and then come back, the filters are persisted but the pills don't appear

So it looks like the only filters that persist are Proxy CDO View. Those should now be regenerating the pills when coming back.

File: src/Components/BidderPortfolio/BidControls/BidControls.jsx [hunk 3]
```diff
@@ -17,44 +21,110 @@ class BidControls extends Component {
...
+    }
+  }
+
+  UNSAFE_componentWillReceiveProps(nextProps) {
+    if (!isEqual(this.props.selection, nextProps.selection)) {
+      if (!(nextProps.selection.length === 1 && get(nextProps, 'selection[0].isCurrentUser'))) {
+        this.setState({ proxyCdos: nextProps.selection }, this.generatePills);
+      }
+    }
+  }
```
Comment (elizabeth-jimenez) [inline src/Components/BidderPortfolio/BidControls/BidControls.jsx:43]: This is to account for our case of wanting Leah AKA current user to remain selected when we clear all filters. But we dont want to show a pill, per say.

So when we hit Clear Filters, the default user is still selected, but the pill is gone. If we intentionally click the Leah pill, however, and she is not the last CDO, then she will be deselected in the dropdown.

**Comment by elizabeth-jimenez** [comment]
😭 Last last things to do are to make sure the updated defaultSort in SelectForm solves the last two checkboxes and a last check of the query params properly updating.

File: src/Components/BidderPortfolio/BidControls/BidControls.jsx (hunk 3 above)
Comment (elizabeth-jimenez) [reply src/Components/BidderPortfolio/BidControls/BidControls.jsx]: Wendy, however, does not come through as isCurrentUser, so she won't be selected.

**Review by mjoyce91 — APPROVED** [review]

**Pull request merged**